tasks/libs/common/gitlab.py

```diff
@@ -56,23 +56,41 @@ def create_pipeline(self, project_name, ref, variables=None):
         data = json.dumps({"ref": ref, "variables": [{"key": k, "value": v} for (k, v) in variables.items()],})
         return self.make_request(path, headers=headers, data=data, json=True)
 
-    def pipelines_for_ref(self, project_name, ref, per_page=100):
+    def all_pipelines_for_ref(self, project_name, ref, sha=None):
         """
-        Gets all pipelines for a given reference
+        Gets all pipelines for a given reference (+ optionally git sha).
+        """
+        pipelines = []
+        page = 1
+
+        # Go through all pages
+        results = self.pipelines_for_ref(project_name, ref, sha=sha, page=page)
+        while results:
+            pipelines.extend(results)
```

Comment:
This is a matter of style but I like more doing `yield from results` and getting rid of `pipelines` now that we can do it (Python 3 only). Wdyt?